src/webview_provider/sharedStateManager.ts

```diff
@@ -0,0 +1,23 @@
+class SharedStateManager {
+  private static instance: SharedStateManager;
+  private listeners: ((message: any) => void)[] = [];
+
+  private constructor() {}
+
+  static getInstance() {
+    if (!SharedStateManager.instance) {
+      SharedStateManager.instance = new SharedStateManager();
+    }
+    return SharedStateManager.instance;
+  }
+
+  addListener(listener: (message: any) => void) {
+    this.listeners.push(listener);
+  }
+
+  postMessage(message: any) {
+    this.listeners.forEach((listener) => listener(message));
+  }
+}
+
+export const sharedStateManager = SharedStateManager.getInstance();
```

Comment:
Why not use vscode event emitters?

Reply:
I will try that

Reply:
I can build event emitters in 2 ways:
- define a static variable in `AltimateWebviewProvider` and listen to the events in sub classes 
- define a service for event emitter, inject it in AltimateWebviewProvider and sub classes and listen to the events

It is required to use same EventEmitter instance in all classes to listen to the fired events.

I prefer to use static variable as eventemitter will not be mutated and keeps it simple. If we go with service, then need to inject in all classes and use it. Lmk your thoughts @mdesmet

Reply:
I think you just need to extract your eventmitter in one class and inject it as a singleton through inversify.